fix: 결제 승인 & 수정 API 수정

Soundbar91 opened the pull request:
# 🔥 연관 이슈

- close #34 

# 🚀 작업 내용
### 결제 승인 API 수정
- 임시 정보로 저장된 결제 관련 데이터(주소, 사장님에게..)와 주문 관련 정보(옵션, 금액..)을 엔티티로 변환하여 DB에 저장합니다.

### 결제 취소 API 수정
- 변경된 엔티티 클래스로 수정했습니다.

# 💬 리뷰 중점사항
- 서비스 쪽만 봐주시면 될 것 같습니다.
- 주문과 결제 관련 테이블에 빠진 필드들이 있어서, 일부 주석처리했습니다. 코인 DB에 반영되는 대로 반영하겠습니다.
- 데이터가 저장이 되는지는 확인을 못했습니다. (결제 DB 트랜잭션과 코인 DB 트랜잭션이 다르고 현재 서비스는 결제 DB 트랜잭션을 사용하고 있어, 아마도 코인 DB에 write를 하면 삑날 가능성이 높아보임 -> 필드 반영하면서 확인하겠습니다.)

Soundbar91 committed:
fix: 결제 승인 응답 dto 필드 수정

- 결제 키, 주문 번호 삭제

Soundbar91 committed:
fix: 결제 승인 API 주석 처리 삭제

Soundbar91 committed:
fix: 주석 처리 롤백

Soundbar91 committed:
fix: 레디스 리파지토리로 변경

Soundbar91 committed:
Merge branch 'develop' of https://github.com/BCSDLab/KOIN_PAYMENT_API into fix/34-payment-confirm-api

Soundbar91 committed:
feat: 주문 관련 리파지토리 추가

Soundbar91 committed:
feat: 결제 관련 엔티티 클래스 추가

Soundbar91 committed:
fix: 결제 승인 API 수정

Soundbar91 committed:
fix: 결제 취소 API 수정

Soundbar91 committed:
fix: 유저 검증 주석 처리 롤백

Soundbar91 committed:
feat: OrderMenu 엔티티 메뉴 옵션 이름 필드 추가

Soundbar91 committed:
feat: 추가된 엔티티 필드 저장 로직 추가

DHkimgit commented on src/main/java/in/koreatech/payment/service/TossService.java:
```diff
@@ -104,24 +119,33 @@ public String createTemporaryTakeoutPayment(String accessToken, TemporaryTakeout
-    //     return payment;
-    // }
+    @Transactional
+    public in.koreatech.koin.domain.order.model.Payment confirmPayment(String accessToken, String paymentKey, String orderId, Integer amount) {
```
`in.koreatech.koin.domain.order.model.Payment` 이거 이렇게 나오는 이유가 있나여?

Soundbar91 replied:
이전에 결제 서버에서 사용하던 Payment랑 `koin.domain.order`하위에 있는 Payment랑 같이 있는 상태에서 수정하다가 import가 잘못된 거 같아요.. 수정하겠읍니다

DHkimgit reviewed (approved):
잘 읽혀요! 말씀해주신 것 처럼 DB 여러 개 사용했을 때 트랜잭션 어떻게 될지 궁금하네요

Soundbar91 committed:
fix: 리뷰 반영 및 버그 수정

- Cart 리파지토리 메소드 수정
- order 엔티티 테이블 어노테이션 설정 수정
- Payment import 수정
- 옵션 메뉴 NPE 해결
- isDeleted 설정 추가

Soundbar91 committed:
feat: 결제 완료 후 장바구니 초기화 로직 추가

Soundbar91 committed:
feat: 결제 승인 메소드 트랜잭션 코인 트랜잭션으로 변경

Soundbar91 committed:
feat: 결제 금액과 요청 금액 검증 로직 추가

Soundbar91 committed:
docs: 스웨거 설명 수정

Soundbar91 committed:
build: Actuator 의존성 추가
